clap_derive/src/derives/arg_enum.rs

```diff
@@ -91,23 +92,32 @@ fn lits(
             }
         })
         .flat_map(|(variant, attrs)| {
-            let mut ret = vec![(attrs.cased_name(), variant.ident.clone())];
+            let mut ret = vec![(attrs.cased_name(), variant.ident.clone(), attrs.clone())];
 
             attrs
                 .enum_aliases()
                 .into_iter()
-                .for_each(|x| ret.push((x, variant.ident.clone())));
+                .for_each(|x| ret.push((x, variant.ident.clone(), attrs.clone())));
 
             ret
         })
         .collect::<Vec<_>>()
 }
 
-fn gen_variants(lits: &[(TokenStream, Ident)]) -> TokenStream {
-    let lit = lits.iter().map(|l| &l.0).collect::<Vec<_>>();
+fn gen_variants(lits: &[(TokenStream, Ident, Attrs)]) -> TokenStream {
+    let lit = lits
+        .iter()
+        .map(|(name, _, attrs)| {
+            let fields = attrs.field_methods_filtered(&["alias"]);
```

Comment:
There is an inconsistency between `Arg` and `ArgValue` that I would consider a bug.  `Arg::alias` is hidden by default and requires `Arg::visible_alias`.  Before, we only had visible alias for `ArgValue` but with this, we can now handle hidden aliases and so `alias` should be `hidden`.  We can add support for `visible_alias` but it doesn't need to be in this PR.

Reply:
makes sense, and then that would mean we probably should move it to `ArgValue`. (it would be possible without, but with the consistency point as well makes more sense to change it.)

I'll probably have a look into it in the next days. Thanks for all the help on these PRs.

Reply:
Thanks for your patience and diligence in following through!  Too many times we, as a community, lose valuable contributions because people don't have the time to polish

Reply:
I'll implemnt only `alias` for now which will be hidden in completions etc.